modules/prediction/container/obstacles/obstacles_container.cc

```diff
@@ -113,34 +121,45 @@ ObstaclesContainer::GetCurrentFramePredictableObstacleIds() const {
 
 void ObstaclesContainer::Clear() {
   obstacles_.Clear();
+  id_mapping_.Clear();
   timestamp_ = -1.0;
 }
 
 
 void ObstaclesContainer::InsertPerceptionObstacle(
     const PerceptionObstacle& perception_obstacle, const double timestamp) {
   // Sanity checks.
-  const int id = perception_obstacle.id();
-  if (id < -1) {
-    AERROR << "Invalid ID [" << id << "]";
+  int prediction_id = perception_obstacle.id();
+  if (curr_frame_id_mapping_.find(prediction_id) !=
+      curr_frame_id_mapping_.end()) {
+    prediction_id = curr_frame_id_mapping_[prediction_id];
+    if (prediction_id != perception_obstacle.id()) {
+      ADEBUG << "Obstacle have got AdaptTracking, with perception_id: "
+             << perception_obstacle.id()
+             << ", and prediction_id: " << prediction_id;
+    }
+  }
+  if (prediction_id < -1) {
+    AERROR << "Invalid ID [" << prediction_id << "]";
```

Comment:
Under which condition the prediction_id will be less than -1?

Reply:
I think in this case, < 0 may be okay

Reply:
-1 is the id of ego vehicle